airflow/jobs.py

```diff
@@ -403,39 +422,77 @@ def schedule_dag(self, dag):
                     dr.end_date = datetime.now()
             session.commit()
 
+            # check last scheduled run
+            # get this dag's max execution date by
+            #  external trigger is false
+            #  and "like" Dag Run ID.ID
+            # bolke: why a like?
             qry = session.query(func.max(DagRun.execution_date)).filter_by(
-                    dag_id = dag.dag_id).filter(
-                        or_(DagRun.external_trigger == False,
-                            # add % as a wildcard for the like query
-                            DagRun.run_id.like(DagRun.ID_PREFIX+'%')))
+                dag_id=dag.dag_id).filter(
+                or_(DagRun.external_trigger is False,
+                    # add % as a wildcard for the like query
+                    DagRun.run_id.like(DagRun.ID_PREFIX+'%')
+                    )
+            )
             last_scheduled_run = qry.scalar()
+            self.logger.debug("Last scheduled run: {}".format(last_scheduled_run))
+
+            # Determine next run date
             next_run_date = None
             if dag.schedule_interval == '@once' and not last_scheduled_run:
                 next_run_date = datetime.now()
```

Comment:
You can add a comment/print here that specifies that we have a `@once` schedule that has not been previously executed (per dag runs) and hence we are scheduling it now.